Error funcs should check for nil error

The `err` passed in may be unexpectedly `nil`
e.g.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xa9dfaf]

goroutine 19 [running]:
testing.tRunner.func1.2({0xc5bd80, 0x132f5f0})
	/opt/google/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/opt/google/go/src/testing/testing.go:1392 +0x39f
panic({0xc5bd80, 0x132f5f0})
	/opt/google/go/src/runtime/panic.go:838 +0x207
github.com/shoenig/test/internal/assertions.EqError({0x0?, 0x0?}, {0xd47273, 0xd})
	/home/shoenig/.go/pkg/mod/github.com/shoenig/test@v0.2.5/internal/assertions/assertions.go:113 +0x2f
github.com/shoenig/test/must.EqError({0xe85900, 0xc00047c1a0}, {0x0, 0x0}, {0xd47273, 0xd})
	/home/shoenig/.go/pkg/mod/github.com/shoenig/test@v0.2.5/must/must.go:47 +0x54
github.com/hashicorp/nomad/nomad/structs.TestServiceCheck_validateNomad.func1(0x0?)
	/home/shoenig/Work/go/nomad/nomad/structs/services_test.go:217 +0x75
```

--- a/internal/assertions/assertions.go
+++ b/internal/assertions/assertions.go
@@ -126,6 +126,10 @@ func Error(err error) (s string) {
 }
 
 func EqError(err error, msg string) (s string) {
+	if err == nil {
+		s = "expected error; got nil"
+		return
+	}
 	e := err.Error()
 	if e != msg {
 		s = "expected matching error strings\n"
@@ -136,6 +140,10 @@ func EqError(err error, msg string) (s string) {
 }
 
 func ErrorIs(err error, target error) (s string) {
+	if err == nil {
+		s = "expected error; got nil"
+		return
+	}
 	if !errors.Is(err, target) {
 		s = "expected errors.Is match\n"
 		s += fmt.Sprintf("↪ error: %v\n", err)

--- a/must/must_test.go
+++ b/must/must_test.go
@@ -62,6 +62,13 @@ func TestEqError(t *testing.T) {
 	EqError(tc, errors.New("oops"), "blah")
 }
 
+func TestEqError_nil(t *testing.T) {
+	tc := newCase(t, `expected error; got nil`)
+	t.Cleanup(tc.assert)
+
+	EqError(tc, nil, "blah")
+}
+
 func TestErrorIs(t *testing.T) {
 	tc := newCase(t, `expected errors.Is match`)
 	t.Cleanup(tc.assert)
@@ -69,6 +76,14 @@ func TestErrorIs(t *testing.T) {
 	e1 := errors.New("foo")
 	e2 := errors.New("bar")
 	ErrorIs(tc, e1, e2)
+}
+
+func TestErrorIs_nil(t *testing.T) {
+	tc := newCase(t, `expected error; got nil`)
+	t.Cleanup(tc.assert)
+
+	err := errors.New("oops")
+	ErrorIs(tc, nil, err)
 }
 
 func TestNoError(t *testing.T) {

--- a/test_test.go
+++ b/test_test.go
@@ -60,6 +60,13 @@ func TestEqError(t *testing.T) {
 	EqError(tc, errors.New("oops"), "blah")
 }
 
+func TestEqError_nil(t *testing.T) {
+	tc := newCase(t, `expected error; got nil`)
+	t.Cleanup(tc.assert)
+
+	EqError(tc, nil, "blah")
+}
+
 func TestErrorIs(t *testing.T) {
 	tc := newCase(t, `expected errors.Is match`)
 	t.Cleanup(tc.assert)
@@ -67,6 +74,14 @@ func TestErrorIs(t *testing.T) {
 	e1 := errors.New("foo")
 	e2 := errors.New("bar")
 	ErrorIs(tc, e1, e2)
+}
+
+func TestErrorIs_nil(t *testing.T) {
+	tc := newCase(t, `expected error; got nil`)
+	t.Cleanup(tc.assert)
+
+	err := errors.New("oops")
+	ErrorIs(tc, nil, err)
 }
 
 func TestNoError(t *testing.T) {

